proposals/NNNN-HID-Support-Plug-in.md

```diff
@@ -0,0 +1,43 @@
+# OEM specific Human Interface Device support as Plug-in architecture in SDL proxy
+
+* Proposal: [SDL-NNNN](NNNN-HID-Support-Plug-in.md)
+* Author: [Jennifer Hodges](https://github.com/jhodges55)
++ Status: Awaiting review
+* Impacted Platforms: [ iOS / Android ]
+
+## Introduction
+
+To support OEM specific HID (ex. Lexus Remote Touch, Mazda Commander Control, BMW Gesture Control, and etc) without new RPC. In the mobile library, plugin architecture + OEM specific plugin module enable OEM specific HID.
+
+## Motivation
+
+Each OEM consider their original UX for safety driving, branding, and easy to operation. In such trend, some OEMs created and will create original specific Human Interface Device (HID). In the past before SDL, proprietary HU applications are implemented by each. And also SDL template architecture can support such specific HID in pre-defined layout properly, because SDL HMI has responsiblity to support HID event to control UI. However, video streaming application which uses NAV_FULLSCREEN_MAP cannot handle such specific HID properly, because there is only OnTouchEvent as RPC. Some devices require not only touch event but also other device specific in/out information. (ex. Haptic feedback, not absolute coordinates but relative, or widget level focus control)
+And, it's not realstic to add new RPC to support every OEM special device (existing device and future device) because SDL core version management and Application vendor effort to support every spec. This proposal is "plugin architecture" to solve "OEM's specific device" without deep discussion to Commonization and Generalization of HID. Each OEM can provide their special device support module quickly with using this architecture. And, it's unnecessary for Application vendors to customize for each OEM's specific device. In other words, Application vendors can avoid customization for each OEM's specific device not only in pre-defined layout (as template) but also in video streaming.
+
+## Proposed solution
+
+Premise condition: "High level UI/Widget library" which support Android/iOS widget for rendering video streaming application in SDL proxy. ex. Android:Virtual Display Encoder, iPhone:SDLCarWindow This library can retrieve High level UI/Widget layout information and create High level UI/Widget event.
+This proposal architecture: Adding "Remote Human Interface Device manager" to SDL proxy (Android/iPhone). This module will provide following functionalities.
+dynamic selection of proper plugin for connected HU. This functionalitiy will be implemented with using VehicleType and reflection of Android Java and iOS ObjectiveC/Swift.
```

Comment:
If these are new sentences, they should start with a capital letter. If these are bullet points, please use `*` before the sentence to add bullet point formatting.